api/src/gameState/gameState.controller.ts

```diff
@@ -38,11 +40,16 @@ export class GameStateController {
   }
 
   @Get('')
-  async getGames(): Promise<{ data: Game[] }> {
+  async getGames(
+    @Query('sort', SortPipe) sort: {},
+    @Query('range', RangePipe) range: {},
+  ): Promise<{ data: Game[] }> {
+    const params = { ...sort, ...range };
     try {
-      const result = await this.gameStateService.getGames({});
+      const result = await this.gameStateService.getGames(params);
       result.data.map((game) => this.gameStateService.deserializerGame(game));
       return result;
+      return;
     } catch (error) {
       throw new NotFoundException('No games was found');
```

Comment:
A couple of things bother me here, first, You catch all error, and throw it as NotFoundException, it may lead to a wrong error message, not good for debugging when needed, also, when you try to get games and there is no data, it's not a NotFound, it's a 200 OK with an empty Array (the result).
One more thing, if you try catch an exception, be sure to not silence it, here, you should at least log it.